retail/interactive-tutorial/RetailSearch.Samples/SearchWithFacetSpec.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_search_product_with_facet_spec]
+
+using Google.Cloud.Retail.V2;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+/// <summary>
+/// Search with facet spec sample class.
+/// </summary>
+public static class SearchWithFacetSpec
+{
+    /// <summary>The project number.</summary>
+    private static readonly string ProjectNumber = Environment.GetEnvironmentVariable("PROJECT_NUMBER");
+
+    /// <summary>The default search placement.</summary>
+    private static readonly string DefaultSearchPlacement = $"projects/{ProjectNumber}/locations/global/catalogs/default_catalog/placements/default_search";
+
+    /// <summary>Get search request.</summary>
+    private static SearchRequest GetSearchRequest(string query, string facetKeyParam)
+    {
+        // Put the intervals here:
+        var facetKey = new SearchRequest.Types.FacetSpec.Types.FacetKey
+        {
+            Key = facetKeyParam
+        };
+
+        var facetSpec = new SearchRequest.Types.FacetSpec
+        {
+            FacetKey = facetKey
+        };
+
+        var searchRequest = new SearchRequest()
+        {
+            Placement = DefaultSearchPlacement, // Placement is used to identify the Serving Config name
+            Query = query,
+            VisitorId = "123456", // A unique identifier to track visitors
+            PageSize = 10
+        };
+
+        searchRequest.FacetSpecs.Add(facetSpec);
```

Comment:
This is more idiomatic:
```suggestion

        var searchRequest = new SearchRequest()
        {
            Placement = DefaultSearchPlacement, // Placement is used to identify the Serving Config name
            Query = query,
            VisitorId = "123456", // A unique identifier to track visitors
            PageSize = 10,
            FacetSpecs = 
            {
                new SearchRequest.Types.FacetSpec
                {
                    FacetKey = new SearchRequest.Types.FacetSpec.Types.FacetKey 
                    {
                         Key = facetKeyParam
                    }
                }
            }
        };
```

Reply:
Resolved